File: tests/integration_tests/tutorial_fr/programs/around_fr.py
# General solution to worlds Around 1 to 4
pense(0)

def tourne_a_droite():
    repeat 3:
        tourne_a_gauche()

def suit_le_mur():
    if rien_a_droite():
        tourne_a_droite()
        avance()
    elif rien_devant():
        avance()
    else:
        tourne_a_gauche()

depose()
while mur_devant():
    tourne_a_gauche()
avance()

while not objet_ici():
    suit_le_mur()

File: tests/integration_tests/tutorial_fr/programs/around_fr.js
// overkill - single program for Around 1, 2, 3, 4
function droite() {
    tourne_a_gauche();
    tourne_à_gauche();  // test accent
    tourne_a_gauche();
}

function suit_le_mur () {
    if (rien_a_droite()) {
        droite();
        avance();
    } else if (rien_devant()) {
        avance();
    } else {
        tourne_a_gauche();
    }
}

depose();
while (!rien_devant()) {
    tourne_a_gauche();
}
avance();

while (!objet_ici()) {
    suit_le_mur();
}
